File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/RotationApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.model.Identity
import my.ssdid.sdk.domain.rotation.KeyRotationManager
import my.ssdid.sdk.domain.rotation.RotationStatus

class RotationApi internal constructor(private val manager: KeyRotationManager) {
    suspend fun prepare(identity: Identity): Result<String> = manager.prepareRotation(identity)
    suspend fun execute(identity: Identity): Result<Identity> = manager.executeRotation(identity)
    suspend fun getStatus(identity: Identity): RotationStatus = manager.getRotationStatus(identity)
}

File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/RotationApi.swift
import Foundation

/// Facade for KERI-inspired key rotation operations.
public struct RotationApi {
    private let manager: KeyRotationManager

    init(manager: KeyRotationManager) {
        self.manager = manager
    }

    public func prepare(identity: Identity) async throws -> String {
        try await manager.prepareRotation(identity: identity)
    }

    public func execute(identity: Identity) async throws -> Identity {
        try await manager.executeRotation(identity: identity)
    }

    public func getStatus(identity: Identity) async -> RotationStatus {
        await manager.getRotationStatus(identity: identity)
    }
}
